File: app/src/main/java/com/autopreference/sample/ApplicationInstance.java
package com.autopreference.sample;

import android.app.Application;
import androidx.annotation.NonNull;

import ru.starksoft.autopreferences.build.AppPreferences;

public class ApplicationInstance extends Application {

	private static ApplicationInstance instance;
	private AppPreferences appPreferences;

	@NonNull
	public static ApplicationInstance getInstance() {
		return instance;
	}

	@NonNull
	public AppPreferences getAppPreferences() {
		return appPreferences;
	}

	@Override
	public void onCreate() {
		super.onCreate();
		appPreferences = new AppPreferences(getApplicationContext());
		instance = this;
	}
}

File: app/src/main/java/com/autopreference/sample/ApplicationInstance.kt
package com.autopreference.sample

import android.app.Application
import ru.starksoft.autopreferences.build.AppPreferences

internal class ApplicationInstance : Application() {

    lateinit var appPreferences: AppPreferences
        private set

    override fun onCreate() {
        super.onCreate()
        appPreferences = AppPreferences(applicationContext)
        instance = this
    }

    companion object {

        lateinit var instance: ApplicationInstance
            private set
    }
}
